src/paste/deploy/loadwsgi.py

```diff
@@ -649,51 +645,37 @@ def get_context(self, object_type, name=None, global_conf=None):
             global_conf or {},
             {},
             self,
-            distribution=pkg_resources.get_distribution(self.spec),
+            distribution=importlib_metadata.distribution(self.spec),
             entry_point_name=ep_name,
         )
 
     def find_egg_entry_point(self, object_type, name=None):
         """
-        Returns the (entry_point, protocol) for the with the given
-        ``name``.
+        Returns the (entry_point, protocol) for with the given ``name``.
         """
         if name is None:
             name = 'main'
+        dist = importlib_metadata.distribution(self.spec)
         possible = []
         for protocol_options in object_type.egg_protocols:
             for protocol in protocol_options:
-                pkg_resources.require(self.spec)
-                entry = pkg_resources.get_entry_info(self.spec, protocol, name)
+                entry = find_entry_point(dist, protocol, name)
                 if entry is not None:
                     possible.append((entry.load(), protocol, entry.name))
                     break
         if not possible:
             # Better exception
-            dist = pkg_resources.get_distribution(self.spec)
             raise LookupError(
-                "Entry point %r not found in egg %r (dir: %s; protocols: %s; "
-                "entry_points: %s)"
+                "Entry point %r not found in egg %r (protocols: %s; entry_points: %s)"
                 % (
                     name,
                     self.spec,
-                    dist.location,
                     ', '.join(_flatten(object_type.egg_protocols)),
                     ', '.join(
-                        _flatten(
-                            [
-                                list(
-                                    (
-                                        pkg_resources.get_entry_info(
-                                            self.spec, prot, name
-                                        )
-                                        or {}
-                                    ).keys()
-                                )
-                                for prot in protocol_options
-                            ]
-                            or '(no entry points)'
-                        )
+                        str(entry)
+                        for prot in protocol_options
+                        for entry in [find_entry_point(dist, prot, name)]
+                        if entry
```

Comment:
this old version has been broken since 2005 as an EntryPoint object never had a keys method